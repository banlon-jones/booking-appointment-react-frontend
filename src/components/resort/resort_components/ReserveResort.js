/* eslint-disable react/jsx-props-no-spreading */
import React, { useState } from 'react';
import { useNavigate, useParams } from 'react-router-dom';
import Modal from 'react-bootstrap/Modal';
import { useForm } from 'react-hook-form';
import { useDispatch } from 'react-redux';
import resortClass from '../ResortDetails.module.css';
// eslint-disable-next-line no-unused-vars
import { createReservation } from '../../../store/reservation/reservation';

function ReserveResort() {
  const { register, handleSubmit } = useForm();
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const [show, setShow] = useState(false);

  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  const { resortId } = useParams();
  const onSubmit = (data) => {
    const payload = { ...data, resort_id: resortId };
    dispatch(createReservation(payload));
    navigate('/reservations');
  };

  return (
    <>
      <button
        type="button"
        className={resortClass.button_container}
        onClick={handleShow}
        data-testid="reserve-button"
      >
        Reserve
      </button>
      <div>
        <Modal
          show={show}
          onHide={handleClose}
          backdrop="static"
          keyboard={false}
        >
          <Modal.Header closeButton>
            <Modal.Title>Make a Reservation</Modal.Title>
          </Modal.Header>
          <Modal.Body>
            <form onSubmit={handleSubmit(onSubmit)}>
              <div className="row">
                <div className="col-12">
                  <p> From </p>
                  {/* eslint-disable-next-line react/jsx-props-no-spreading */}
                  <input
                    type="date"
                    id="from"
                    className="form-control"
                    {...register('date_from')}
                  />
                </div>
                <div className="col-12">
                  <p> To </p>
                  {/* eslint-disable-next-line react/jsx-props-no-spreading */}
                  <input
                    type="date"
                    className="form-control"
                    {...register('date_to')}
                  />
                </div>
              </div>
              <div className="m-3">
                <button type="submit" className="btn btn-success">
                  {' '}
                  submit
                </button>
              </div>
            </form>
          </Modal.Body>
        </Modal>
      </div>
    </>
  );
}

export default ReserveResort;
