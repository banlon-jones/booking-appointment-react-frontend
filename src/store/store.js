/* eslint-disable import/no-named-as-default */
/* eslint-disable import/no-named-as-default-member */
import { configureStore } from '@reduxjs/toolkit';
import userSlice from './user/user';
import resortDetailSlice from './resort-details/resortDetail';

const store = configureStore({
  reducer: {
    user: userSlice,
    resortDetail: resortDetailSlice,
  },
});

export default store;
