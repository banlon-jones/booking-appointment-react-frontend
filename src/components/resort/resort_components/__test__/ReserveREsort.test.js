/* eslint-disable comma-dangle */
import React from 'react';
import { cleanup, screen, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../../../store/store';
import ReserveResort from '../ReserveResort';

afterEach(cleanup);

describe('React Test for AddResort', () => {
  test('Reserve Resort Test', () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <ReserveResort />
        </Provider>
      </BrowserRouter>
    );
    const ResortElement = screen.getByTestId('reserve-button');
    expect(ResortElement).toBeInTheDocument('resort');
  });
});

describe('Reserve Resort Snapshot', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <ReserveResort />
          </Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
