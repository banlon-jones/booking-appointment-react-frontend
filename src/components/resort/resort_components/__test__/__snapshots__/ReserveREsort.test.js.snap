// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reserve Resort Snapshot renders correctly 1`] = `
Array [
  <button
    className="button_container"
    data-testid="reserve-button"
    onClick={[Function]}
    type="button"
  >
    Reserve
  </button>,
  <div />,
]
`;
